src/sdl_ui.cpp

```diff
@@ -419,6 +419,90 @@ bool SdlUi::ShowCursor(bool flag) {
 void SdlUi::Blit2X(Bitmap const& src, SDL_Surface* dst_surf) {
 	if (SDL_MUSTLOCK(dst_surf)) SDL_LockSurface(dst_surf);
 
+#if defined(__MORPHOS__) || defined(__amigaos4__)
+	// Quick & dirty big endian 2x zoom blitter
```

Comment:
May I ask what the problem with the normal Blit2x code is?

Reply:
It's slow, I barely got 20-25 fps on a Mac Mini G4 with it. With this one I'm getting a relatively stable 50 fps on the same machine. It's also fast enough to get 50 fps in WinUAE in the 2x zoom mode.

Reply:
https://github.com/EasyRPG/Player/blob/4bebc3d6dbbf0e297e04aa8188835a8fb49c66d5/src/bitmap.cpp#L1081-L1094
...could be why (arbitrary matrix transform? theoretically fastpathable but at some point the fastpath checks outweigh the benefits)
There doesn't seem to be anything in pixman to do such stretching - at best one could copy columns twice, then rows within the target buffer starting from the bottom (so it'll only overwrite already handled stuff)

Reply:
Is the system running with 16 bit color scheme?
For some reason the blitting code of Pixman is extremely slow (at least twice) when operating on 16bit (also on little encian). Is probably our fault because this makes no real sense but nobody really understands the undocumented pixman API ^^'.
On the Wii we use a custom SDL build which forces a 32bit surface and this gives a gigantic speedup, maybe is the same problem here.

Reply:
Ghabry: For that particular case: Is it possible the source image surfaces are 32-bit but the destinations are 16 (necessitating conversion)?

Reply:
In my tests even 16-16 was slow according to the profiler.
There are many non-obvious snail-paths in pixman, e.g. when you use ARGB instead of RGBA the code is suddenly 50% slower (could remember this wrong but at least some 32-bit pixel formats are much slower than others).

Reply:
Not really happy with this either, however, is SDL1.2 only → does not really matter.
Maybe we could even use this under Wii in the meanwhile...

Reply:
Also, maybe we should hide this stuff deeply in some `src/platform/amiga/port.cpp`... :P